mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java

```diff
@@ -311,22 +340,95 @@ protected void onCreate(Bundle savedInstanceState) {
+        }
 
-            //  Launch app intro
-            final Intent i = new Intent(OpenHABMainActivity.this, IntroActivity.class);
-            startActivity(i);
+        final String url = "/rest/bindings";
+        conn.getAsyncHttpClient().get(url, new MyHttpClient.TextResponseHandler() {
+            @Override
+            public void onFailure(Call call, int statusCode, Headers headers, String responseString, Throwable throwable) {
+                mOpenHABVersion = 1;
+                Log.d(TAG, "openHAB version 1 - got error " + throwable + " accessing " + url);
+                conn.getAsyncHttpClient().addHeader("Accept", "application/xml");
+                selectSitemap();
+            }
 
-            prefsEdit.putBoolean("firstStart", false).apply();
+            @Override
+            public void onSuccess(Call call, int statusCode, Headers headers, String responseString) {
+                mOpenHABVersion = 2;
+                Log.d(TAG, "openHAB version 2");
+                selectSitemap();
+            }
+        });
+    }
+
+    private void discoverOpenHAB() {
+        if (mServiceResolver != null && mServiceResolver.isAlive()) {
+            Log.d(TAG, "openHAB is already being discovered, another start of discovery refused.");
+            return;
         }
+        mProgressDialog = ProgressDialog.show(this, "",
+                getString(R.string.info_discovery), true);
 
-        OnUpdateBroadcastReceiver.updateComparableVersion(prefsEdit);
-        prefsEdit.apply();
+        mServiceResolver = new AsyncServiceResolver(
+                this, this, openHABServiceType);
+        mServiceResolver.start();
+    }
+
+    @Override
+    public void onServiceResolved(ServiceInfo serviceInfo) {
+        stopProgressDialog();
+
+        Log.d(TAG, "Service resolved: "
+                + serviceInfo.getHostAddresses()[0]
+                + " port:" + serviceInfo.getPort());
+        String openHABUrl = "http://" + serviceInfo.getHostAddresses()[0] + ":" +
+                String.valueOf(serviceInfo.getPort()) + "/";
+
+        PreferenceManager.getDefaultSharedPreferences(this).edit().putString(Constants.PREFERENCE_LOCAL_URL, openHABUrl).apply();
+
+        restartAfterSettingsUpdate();
+    }
+
+    @Override
+    public void onServiceResolveFailed() {
+        stopProgressDialog();
+        PreferenceManager
+                .getDefaultSharedPreferences(this)
+                .edit()
+                .putBoolean(Constants.PREFERENCE_DEMOMODE, true)
+                .apply();
+        restartAfterSettingsUpdate(true);
+    }
+
+    private void stopProgressDialog() {
+        try {
+            mProgressDialog.dismiss();
+            mProgressDialog = null;
+        } catch (Exception e) {
+            // This is to catch "java.lang.IllegalArgumentException: View not attached to window manager"
+            // exception which happens if user quited app during discovery
+        }
```

Comment:
Better check instead of catching the (too generic) exception:

```
if (!isFinishing() && mProgressDialog != null && mProgressDialog.isShowing()) {
    mProgressDialog.dismiss();
}
mProgressDialog = null;
```

Reply:
Done!